module.c

```diff
@@ -59,6 +59,15 @@ static void teardown_time_limit(ContextData *context) {
 	}
 }
 
+// Converts a JSValue to a C string.
+// Can return NULL. Deallocate with JS_FreeCString.
+static const char* js_to_cstring(JSContext* context, JSValue value) {
+	JSValue string = JS_ToString(context, value);
```

Comment:
Test for JS_EXCEPTION

Reply:
OK. (This function returned NULL for exceptions before as well)

Reply:
Sorry, I didn't mean the check the input argument for being an exception (I think that's the caller's responsibility, exception checking should be done just after calling an API, exception values should not be passed around at all ideally), but rather `string` being an exception. Stringification can fail, it calls toString for instance.

Reply:
Makes perfect sense.